### PR TITLE
Feature release/cdap 16308 preview perf

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java [hunk 1]
```diff
@@ -19,15 +19,20 @@
...
   private final ProgramId program;
-  private final AppRequest<T> appRequest;
+  private final AppRequest<?> appRequest;
+
+  public PreviewRequest(ProgramId program) {
+    this(program, null);
+  }
 
-  public PreviewRequest(ProgramId program, AppRequest<T> appRequest) {
+  public PreviewRequest(ProgramId program, AppRequest<?> appRequest) {
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java]: appRequest should be marked Nullable

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java [hunk 2]
```diff
@@ -19,15 +19,20 @@
...
 import io.cdap.cdap.proto.id.ProgramId;
 
+import javax.annotation.Nullable;
+
 /**
  * Represents the preview application request.
- * @param <T> the type of application config
  */
-public class PreviewRequest<T> {
+public class PreviewRequest {
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java:27]: Can you explain a bit why make PreviewRequest non-parameterized will help the preview perf?

**Review by yaojiefeng — APPROVED** [review]
just one question and one minor comment, LGTM

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java (hunk 2 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java:27]: It doesn't. But it is never get used. So it is useless to be parameterized.

**Comment by chtyim** [comment]
https://builds.cask.co/browse/CDAP-RUT1738

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java (hunk 1 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequest.java]: fixed

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-RUT1738

**Pull request merged**